pkg/conditions/node/resolve.go

```diff
@@ -61,6 +63,12 @@ func resolve(node Node, timeout time.Duration) (Node, error) {
 		hostname := hostname
 		go func() {
 			addrs, err := net.LookupHost(hostname)
+
+			// always arrange the addresses that IPv4 comes before IPv6
+			sort.SliceStable(addrs, func(i, j int) bool {
+				return strings.Contains(addrs[i], ".") && !strings.Contains(addrs[i], ":") && strings.Contains(addrs[j], ":")
```

Comment:
TODO: there should be a more proper way of determining IPv4 vs. IPv6. I'm not sure if there aren't cases for IPv6, where this fails.

Reply:
You could parse it as IP address and then assert `ip.To4() != nil` (in which case it would be an IPv4 address). However, there's some caveats, see e.g. here: https://stackoverflow.com/questions/22751035/golang-distinguish-ipv4-ipv6

Reply:
Hah. Funny. The `::FFFF:127.0.0.1` case is exactly what I was thinking about :). Your proposal sounds good enough though. I'll adjust.